### DISCUSSION TITLE
Possible to use specs within pairs?

### DISCUSSION THREAD

**Discussion by adammsteele** [dabapps/django-readers#77] [Q&A]
Hi (again),

When defining a spec, you can drop down a level to pairs for more control, but once you have done that, are you stuck at the pairs level within that key, or is there some way to start using spec notation again?

For example, imagine an author detail spec, where we need to provide a custom queryset to a relationship:

```
def get_spec(self):
    return [
        'name',
        {
            'books_for_user' : pairs.reverse_relationship(
                'book_set',
                'author',
                Book.objects.for_user(self.request.user),
                # I'd like to revert back to spec notation here, but it doesn't seem possible
                pairs.combine(
                    pairs.producer_to_projector(
                        "title",
                        pairs.field("title"),
                    ),
                )
        )}
    ]
```

Just wanted to check if I was missing anything, but it doesn't seem possible since pairs don't know anything about specs.

Thanks,
Adam

**Comment by j4mie** [comment]
Hi @adammsteele 

So sorry for the crazy delay in replying to this. I'm not sure how I missed it - I need to look at my notification settings for discussions 😬 

You might've solved this already, but I think what you're trying to do isn't quite the right approach. Are you trying to filter your related `book_set` based on the user? I think what you want to do then is something more like:

```python
def get_spec(self):
    return [
        'name',
        {
            'books_for_user' : {
                "book_set": [
                    (lambda queryset: queryset.for_user(self.request.user), projectors.noop)
                    "title",
                ],
            }
        }
    ]
```

In other words, you want to filter the `book_set` prefetch queryset, rather than override it entirely. Does that make sense? Sorry if I've misunderstood!
